src/ntpclient.cpp

```diff
@@ -30,74 +27,57 @@ bool CNtpClient::getTimestamp(uint64_t &timeRecv)
         ip::udp::socket socket(io_service);
         socket.open(ip::udp::v4());
 
-        boost::array<unsigned char, 48> sendBuf = {{010,0,0,0,0,0,0,0,0}};
+        boost::array<unsigned char, 48> sendBuf = {{010, 0, 0, 0, 0, 0, 0, 0, 0}};
 
         socket.send_to(boost::asio::buffer(sendBuf), receiver_endpoint);
 
         boost::array<uint32_t, 1024> recvBuf;
         ip::udp::endpoint sender_endpoint;
 
-        try
-        {
-
+        try {
             fd_set fileDescriptorSet;
             struct timeval timeStruct;
 
             timeStruct.tv_sec = GetArg("-ntptimeout", DEFAULT_NTP_TIMEOUT);
             timeStruct.tv_usec = 0;
             FD_ZERO(&fileDescriptorSet);
 
-            int nativeSocket = socket.native();
-
-            FD_SET(nativeSocket,&fileDescriptorSet);
+            #if BOOST_VERSION >= 104700
```

Comment:
Why are we now detecting for the boost version

Reply:
@aguycalled  found this fix for me, I believe it's a fix for a known bug in boost